components/org.wso2.micro.integrator.log4j2.plugins/src/main/java/org/wso2/micro/integrator/log4j/plugins/LogMaskConverter.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 WSO2, Inc. (http://wso2.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.micro.integrator.log4j.plugins;
+
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.pattern.ConverterKeys;
+import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
+import org.wso2.micro.integrator.core.util.MicroIntegratorBaseUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Plugin(name = "LogMaskConverter", category = "Converter")
+@ConverterKeys({"mm"})
+/**
+ * Log Masking converter to mask logs.
+ * Converter key mm used to change existing message pattern in log4j2.properties
+ */
+public class LogMaskConverter extends LogEventPatternConverter {
+
+    private static final LogMaskConverter INSTANCE = new LogMaskConverter();
+    private static final String DEFAULT_MASKING_PATTERNS_FILE_NAME = "wso2-log-masking.properties";
+    private static final String REPLACEMENT_STRING = "*****";
+    private static final String MASK_PATTERN = "mm";
+
+    private final List<Pattern> logMaskingPatterns;
+    private boolean isMaskAvailable = false;
+
+    public static LogMaskConverter newInstance(String[] options) {
```

Comment:
What is the purpose of passing options here?

Reply:
This function used as log4j extended functionality through its interfaces. Since we don't need options, we can ignore those.